src/library/scala/collection/immutable/SeqMap.scala

```diff
@@ -16,19 +16,19 @@ package immutable
 
 import scala.collection.mutable.{Builder, ReusableBuilder}
 
-/**
-  * A generic trait for ordered immutable maps. Concrete classes have to provide
-  * functionality for the abstract methods in `SeqMap`.
-  *
-  * Note that when checking for equality [[SeqMap]] does not take into account
-  * ordering.
-  *
-  * @tparam K      the type of the keys contained in this linked map.
-  * @tparam V      the type of the values associated with the keys in this linked map.
-  *
-  * @define coll immutable seq map
-  * @define Coll `immutable.SeqMap`
-  */
+/** A generic trait for ordered immutable maps. Concrete classes have to provide
+ *  functionality for the abstract methods in `SeqMap`.
+ *
+ *  Methods that return a new map, such as [[removed]] and [[updated]], must preserve ordering.
```

Comment:
personal opinion, it should describe its behaviour to users, not its contract to implementors

```suggestion
 *  Methods that return a new map, such as [[removed]] and [[updated]], preserve ordering.
```

Reply:
The sentence follows, "Concrete classes have to provide functionality for the abstract methods in SeqMap." (Arguably that is talking down to whomever it addresses.) I could tweak "have to" and "must" to "shall", which sounds like a contract for all parties to it. And add, "No other warranties, express or implied, etc, etc." I could make the ordering clause a "Note"; not sure if scaladoc has `@implNote`. I take your point, but I don't know what to do. I will look for examples of prior art. If I find "none" (nun), then that would be "priory" art.